### ISSUE TITLE
Get an error when using pluck() method

### ISSUE THREAD

**Issue by francoislevesque** [jeromegamez/typed-collection#4]
I currently get an exception (A XXXXX collection only accepts objects of the following type(s): XXXXX.) when using the `pluck()` method on a typed collection. I must do `untype()` before. 

Is this an intended behavior?

**Comment by jeromegamez** [comment]
Not as far as I know... could you show me a failing code example so that I can easier reproduce the issue, please? I‘ll then implement the fix.

**Comment by jeromegamez** [comment]
I quickly checked this, and got the same error. It then got me thinking: how would you expect it to work? What to pluck? Could you give me a use case?

**Comment by francoislevesque** [comment]
I did the following test:
```
<?php

namespace Tests\Unit;

use Gamez\Illuminate\Support\TypedCollection;
use Tests\TestCase;

class TypedCollectionTest extends TestCase
{
    public function test_pluck()
    {
        $collection = new FooCollection([
            new Foo(1, "a"),
            new Foo(2, "b"),
            new Foo(3, "c")
        ]);

        $collection->pluck('name'); // Should return array of null
        $collection->pluck('id'); // Should return array of ids
    }
}

class Foo 
{
    public $id;
    private $name;

    public function __construct($id, $name)
    {
        $this->id = $id;
        $this->name = $name;
    }

    public function getName()
    {
        return $this->name;
    }
}

class FooCollection extends TypedCollection
{
    protected static $allowedTypes = [Foo::class];
}
```

Im my opinion, pluck should work like a normal collection: try to get the attribute on the object if it exists, else return null.

```
$collection = collect([
   new Foo(1, "a"),
   new Foo(2, "b"),
   new Foo(3, "c")
]);

$collection->pluck('name');  // returns [null,null,null]
$collection->pluck('id');  // returns [1,2,3]
```

**Comment by jeromegamez** [comment]
Thank you for the detailed description! I'll get right onto it 💪

**Comment by jeromegamez** [comment]
The fix is released with version [2.2.1](https://github.com/jeromegamez/typed-collection/releases/tag/2.2.1). Thank you for bringing this up! 🌺
